### PR TITLE
Add strategy to NewMarket & Fix typo

### LINKED ISSUE #706: [v1] Add MarketStrategy to NewMarket request
NewMarket should take an additional MarketStrategy parameter so that client can set it in market creation flow.
If Pluggable is chosen then price should be set to zero.

### PR DESCRIPTION
**Pull request by altafan**
This allows the operator to define the market strategy at creation time.

Closes #706.

Please @sekulicd review this.

### CODE REVIEW

File: internal/interfaces/grpc/handler/utils.go [hunk 1]
```diff
@@ -57,14 +57,25 @@ func parseMarketFee(fee *tdexv2.MarketFee) (int64, int64, error) {
 	return baseFee, quoteFee, nil
 }
 
+func parseStrategySafe(strategyType daemonv2.StrategyType) (int, error) {
```
Comment (sekulicd) [inline internal/interfaces/grpc/handler/utils.go:60]: can we have one method that is parsing StrategyType?

File: internal/interfaces/grpc/handler/utils.go [hunk 2]
```diff
@@ -57,14 +57,25 @@ func parseMarketFee(fee *tdexv2.MarketFee) (int64, int64, error) {
...
+	strategy, err := parseStrategy(strategyType)
+	if err != nil {
+		return -1, err
+	}
+	if strategy == domain.StrategyTypeUndefined {
```
Comment (sekulicd) [inline internal/interfaces/grpc/handler/utils.go:65]: case domain. StrategyTypeUndefined is never returned by the parseStrategy function?

File: internal/core/domain/market.go [hunk 3]
```diff
@@ -104,6 +104,12 @@ func NewMarket(
 	if !isValidPrecision(quoteAssetPrecision) {
 		return nil, ErrMarketInvalidQuoteAssetPrecision
 	}
+	if !isValidStrategy(strategyType) {
+		return nil, ErrMarketUnknownStrategy
+	}
+	if strategyType == StrategyTypeUndefined {
+		strategyType = StrategyTypeBalanced
```
Comment (sekulicd) [inline internal/core/domain/market.go:111]: here u set type to StrategyTypeBalanced if "undefined/unspecified" and in handler/utils.go if it is StrategyType_STRATEGY_TYPE_UNSPECIFIED u set type to StrategyTypePluggable?

File: internal/interfaces/grpc/handler/utils.go (hunk 2 above)
Comment (altafan) [reply internal/interfaces/grpc/handler/utils.go:65]: there was already an exisitng `parseStrategy` function taking care of converting a StrategyType used by `UpdateMarketStrategy` RPC. In this RPC `StrategyTypeUndefined` is not a valid strategy, while it is in `NewMarket` (we default to balanced strategy if it is not specified by the user)

File: internal/interfaces/grpc/handler/utils.go (hunk 1 above)
Comment (altafan) [reply internal/interfaces/grpc/handler/utils.go:60]: https://github.com/tdex-network/tdex-daemon/pull/708#discussion_r1183576891

File: internal/core/domain/market.go (hunk 3 above)
Comment (altafan) [reply internal/core/domain/market.go:111]: https://github.com/tdex-network/tdex-daemon/pull/708#discussion_r1183576891

**Review by sekulicd — APPROVED** [review]

**Pull request merged**